Scripts/RemediationScripts/Remediate-SetStorageAccountMinReqTLSVersion.ps1

```diff
@@ -0,0 +1,959 @@
+    Write-Host $([Constants]::SingleDashLine)
+    if ($PerformPreReqCheck)
+    {
+        try
+        {
+            Write-Host "Setting up prerequisites..."
+            Setup-Prerequisites
+        }
+        catch
+        {
+            Write-Host "Error occurred while setting up prerequisites. Error: $($_)" -ForegroundColor $([Constants]::MessageType.Error)
+            break
+        }
+    }
+
+    # Connect to Azure account
+    $context = Get-AzContext
+
+    if ([String]::IsNullOrWhiteSpace($context))
+    {
+        Write-Host "No active Azure login session found. Exiting..." -ForegroundColor $([Constants]::MessageType.Error)
+        Write-Host $([Constants]::DoubleDashLine)
+        break
+    }
+
+    # Setting up context for the current Subscription.
+    $context = Set-AzContext -SubscriptionId $SubscriptionId -ErrorAction Stop
+    
+    Write-Host $([Constants]::SingleDashLine)
+    Write-Host "Subscription Name: $($context.Subscription.Name)"
+    Write-Host "Subscription ID: $($context.Subscription.SubscriptionId)"
+    Write-Host "Account Name: $($context.Account.Id)"
+    Write-Host "Account Type: $($context.Account.Type)"
+    Write-Host $([Constants]::SingleDashLine)
+
+    Write-Host "*** To reset TLS Versions for Storage Account(s) in a Subscription, Contributor or higher privileges on the Storage Account(s) are required. ***" -ForegroundColor $([Constants]::MessageType.Info)
+    Write-Host $([Constants]::SingleDashLine)
+    Write-Host "`n[Step 2 of 4] Preparing to fetch all Storage Account(s)..."
+    Write-Host $([Constants]::SingleDashLine)
+    if(-not (Test-Path -Path $FilePath))
+    {
+        Write-Host "ERROR: Input file - $($FilePath) not found. Exiting..." -ForegroundColor $([Constants]::MessageType.Error)
+        Write-Host $([Constants]::DoubleDashLine)
+        break
+    }
+
+    Write-Host "Fetching all Storage Account(s) from $($FilePath)" -ForegroundColor $([Constants]::MessageType.Info)
+    Write-Host $([Constants]::SingleDashLine)
+       
+    $storageAccountsFromFile = Import-Csv -LiteralPath $FilePath
+    $validStorageAccounts = $storageAccountsFromFile | Where-Object { ![String]::IsNullOrWhiteSpace($_.StorageAccountName) }
+        
+    $StorageAccounts = @()
+    $StorageAccountList = @()
+
+    $validStorageAccounts | ForEach-Object {
+        $storageAccount = $_
+        $storageAccountName = $_.StorageAccountName
+        $resourceGroupName = $_.ResourceGroupName
+        $minimalTlsVersionBefore = $_.MinimalTlsVersionBefore
+        $minimalTlsVersionAfter = $_.MinimalTlsVersionAfter
+
+        try
+        {
+            $StorageAccount = ( Get-AzStorageAccount -StorageAccountName $storageAccountName  -ResourceGroupName $resourceGroupName -ErrorAction SilentlyContinue) 
+            $StorageAccountTls = (Get-AzStorageAccount -Name $storageAccountName -ResourceGroupName $resourceGroupName).MinimumTlsVersion
```

Comment:
Two Get calls again. Confirm once whether this is needed.

Reply:
Removed. Not required with new logic